core/domain/learner_progress_services_test.py

```diff
@@ -481,6 +507,89 @@ def test_get_all_completed_exp_ids(self):
         self.assertEqual(learner_progress_services.get_all_completed_exp_ids(
             self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
+    def test_get_filtered_completed_exp_summaries(self):
+        # Add explorations to the completed list.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_0)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_1)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_3)
+        self.assertEqual(
+            len(learner_progress_services.get_all_completed_exp_ids(
+                self.user_id)), 3)
```

Comment:
This looks great now, can we also check the ids of each exploration like you have done in line 539? (Sorry if my last comment related to this was not clear.)

Reply:
Yep, definitely!